pkg/tests/mappings/mappings_test.bzl

```diff
@@ -592,6 +608,269 @@ def _test_pkg_mkdirs():
+        "pkg_filegroup file list does not match expectations",
+    )
+    if ctx.attr.verify_origins:
+        asserts.equals(
+            env,
+            [t.label for t in ctx.attr.expected_pkg_files],
+            [origin for _, origin in target_under_test[PackageFilegroupInfo].pkg_files],
+            "pkg_filegroup file origin list does not match expectations",
+        )
+
+    asserts.equals(
+        env,
+        [t[PackageDirsInfo] for t in ctx.attr.expected_pkg_dirs],
+        [info for info, _ in target_under_test[PackageFilegroupInfo].pkg_dirs],
+        "pkg_filegroup directory list does not match expectations",
+    )
+    if ctx.attr.verify_origins:
+        asserts.equals(
+            env,
+            [t.label for t in ctx.attr.expected_pkg_dirs],
+            [origin for _, origin in target_under_test[PackageFilegroupInfo].pkg_dirs],
+            "pkg_filegroup directory origin list does not match expectations",
+        )
+
+    asserts.equals(
+        env,
+        [t[PackageSymlinkInfo] for t in ctx.attr.expected_pkg_symlinks],
+        [info for info, _ in target_under_test[PackageFilegroupInfo].pkg_symlinks],
+        "pkg_filegroup symlink map does not match expectations",
+    )
+    if ctx.attr.verify_origins:
+        asserts.equals(
+            env,
+            [t.label for t in ctx.attr.expected_pkg_symlinks],
+            [origin for _, origin in target_under_test[PackageFilegroupInfo].pkg_symlinks],
+            "pkg_filegroup symlink origin list does not match expectations",
+        )
+
+    # Verify that the DefaultInfo is propagated properly out of the input
+    # pkg_files's -- these are the files that need to be passed along to the
+    # packaging rules.
+    expected_packaged_files = sorted(_flatten([
+        t[DefaultInfo].files.to_list()
+        for t in ctx.attr.expected_pkg_files
+    ]))
+    packaged_files = sorted(target_under_test[DefaultInfo].files.to_list())
+
+    asserts.equals(
+        env,
+        expected_packaged_files,
+        packaged_files,
+        "pkg_filegroup does not propagate DefaultInfo from pkg_file inputs",
+    )
+
+    return analysistest.end(env)
+
+pkg_filegroup_contents_test = analysistest.make(
+    _pkg_filegroup_contents_test_impl,
+    attrs = {
+        "expected_pkg_files": attr.label_list(
+            providers = [PackageFilesInfo],
+            default = [],
+        ),
+        "expected_pkg_dirs": attr.label_list(
+            providers = [PackageDirsInfo],
+            default = [],
+        ),
+        "expected_pkg_symlinks": attr.label_list(
+            providers = [PackageSymlinkInfo],
+            default = [],
+        ),
+        "verify_origins": attr.bool(
+            default = True,
+        ),
+    },
+)
+
+def _test_pkg_filegroup(name):
+    pkg_files(
+        name = "{}_pkg_files".format(name),
+        srcs = ["foo", "bar"],
+        prefix = "bin",
+        tags = ["manual"],
```

Comment:
Why the manual tag?

Reply:
It's to prevent the rules from actually being built when you run `bazel test //...`.

I'm not really sure about how analysis tests work, but if the rules aren't completely executed, then the labels used as sources can be bogus.  They'll still be presented to providers in the dependent rules, and the whole thing "just works" and I don't have to create a bunch of extra files.

It's a similar pattern to what's used when you write [negative analysis tests](https://docs.bazel.build/versions/master/skylark/testing.html#failure-testing).

Reply:
An interesting idea from @beasleyr-vmw that we've been mulling over internally is an extension to the tagging system where you could embed a reason into a tag (something like `manual:analysis-test`).

In particular, the `manual` tag has many use cases and it's usually not clear what, and comments are hard to query.  Something like this could make intent clearer w/o introducing additional terminology.

Probably worth a ticket against bazel itself if there isn't one already.